terraform/cp_cloud_function_files.sh

```diff
@@ -0,0 +1,6 @@
+set -e
```

Comment:
Do folks have thoughts on how to integrate this more with the terraform? Right now, my process is to do
```
./cp_cloud_function_files.sh && terraform apply
```
to deploy new code to the cloud functions. It'd be rad if it was just `terraform apply`

Reply:
I wonder if you can use `repo_source` somehow. I don't know much about cloud functions though. https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/cloudfunctions2_function#repo_source

Reply:
We aren't working with Cloud Source Repositories, so I don't think we can use `repo_source`

Reply:
I'm fine with you submitting as is-- I don't know much about cloud functions, but it seems strange that we have to run a script before `terraform apply`. But we can change this later on too.

Reply:
> I'm fine with you submitting as is-- I don't know much about cloud functions, but it seems strange that we have to run a script before `terraform apply`. But we can change this later on too.

I specifically asked if folks had ideas on how to get rid of that `cp_cloud_function_files.sh` script and incorporate it into terraform!

`data "archive_file"` is a bit restrictive. The script is just copying files to make them easier for terraform to pull into a zip via `data "archive_file"`